native/cocos/math/Vec4.cpp

```diff
@@ -329,10 +331,12 @@ const Vec4 Vec4::UNIT_Y = Vec4(0.0F, 1.0F, 0.0F, 0.0F);
 const Vec4 Vec4::UNIT_Z = Vec4(0.0F, 0.0F, 1.0F, 0.0F);
 const Vec4 Vec4::UNIT_W = Vec4(0.0F, 0.0F, 0.0F, 1.0F);
 
+#if CC_PLATFORM != CC_PLATFORM_EMSCRIPTEN
 template <>
 ccstd::hash_t Hasher<Vec4>::operator()(const Vec4 &v) const {
     return ccstd::hash_range(reinterpret_cast<const uint64_t *>(&v.x),
                              reinterpret_cast<const uint64_t *>(&v.x + 4));
 }
+#endif
```

Comment:
wasm don't need this function?

Reply:
Only the basic functions of math are used, not the hash function。

Reply:
Got it. But new codes added in future may not compile in wasm mode. May be we should add a CI to make sure codes always work in wasm mode.

Reply:
And where to define `CC_PLATFORM_EMSCRIPTEN `?

Reply:
> Got it. But new codes added in future may not compile in wasm mode. May be we should add a CI to make sure codes always work in wasm mode.

We already have a wasm ci to check compiling result when building wasm.